src/ngraph_backend_manager.cc

```diff
@@ -107,7 +109,28 @@ bool BackendManager::IsSupportedBackend(const string& backend_name) {
     return false;
   }
   return true;
-};
+}
+
+Status BackendManager::GetCurrentlySetBackendName(string* current_backend) {
+  string possible_backend;
+  const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+  // Environment variable has higher priority than API set value
+  if (ng_backend_env_value != nullptr) {
+    possible_backend = std::string(ng_backend_env_value);
+  } else {
+    possible_backend = BackendManager::ng_backend_name_;
```

Comment:
We dont need to check if the possible_backend is empty or is_supported in the case when NGRAPH_TF_BACKEND is nullptr. As we are doing that check in SetBackendName().

I was actually working on something similar
https://github.com/tensorflow/ngraph-bridge/pull/53/files#diff-73f0d2f88e27f31832d455ac7a525189R112